src/libConsensus/ConsensusLeader.cpp

```diff
@@ -236,32 +398,37 @@ bool ConsensusLeader::ProcessMessageCommitFailure(
 }
 
 bool ConsensusLeader::GenerateChallengeMessage(vector<unsigned char>& challenge,
-                                               unsigned int offset) {
+                                               unsigned int offset,
+                                               uint16_t subsetID) {
   LOG_MARKER();
 
   // Generate challenge object
   // =========================
 
+  ConsensusSubset& subset = m_consensusSubsets.at(subsetID);
+
   // Aggregate commits
-  CommitPoint aggregated_commit = AggregateCommits(m_commitPoints);
+  CommitPoint aggregated_commit = AggregateCommits(subset.m_commitPoints);
   if (!aggregated_commit.Initialized()) {
-    LOG_GENERAL(WARNING, "AggregateCommits failed");
+    LOG_GENERAL(WARNING, "[Subset " << subsetID << "] AggregateCommits failed");
     m_state = ERROR;
```

Comment:
I'm thinking maybe the errors here (and perhaps elsewhere) have to be re-evaluated.  Do we really want to set the overall `m_state` to `ERROR`, or set just `subset.m_state` to `ERROR`?